src/model_internal.jl

```diff
@@ -0,0 +1,83 @@
+@scoped_enum(
+    RunStatus,
+    NOT_READY = -2,
+    READY = -1,
+    SUCCESSFUL = 0,
+    RUNNING = 1,
+    FAILED = 2,
+)
+
+struct TimeSeriesCacheKey
+    component_uuid::Base.UUID
+    time_series_type::Type{<:TimeSeriesData}
+    name::String
+    multiplier_id::Int
+end
+
+mutable struct SimulationInfo
+    number::Int
+    sequence_uuid::Base.UUID
+end
+
+mutable struct ModelInternal{T <: AbstractOptimizationContainer}
+    container::T
+    ic_model_container::Union{Nothing, T}
+    status::BuildStatus
+    run_status::RunStatus
+    base_conversion::Bool
+    executions::Int
+    execution_count::Int
+    output_dir::Union{Nothing, String}
+    simulation_info::Union{Nothing, SimulationInfo}
+    time_series_cache::Dict{TimeSeriesCacheKey, <:TimeSeriesCache}
+    recorders::Vector{Symbol}
+    console_level::Base.CoreLogging.LogLevel
+    file_level::Base.CoreLogging.LogLevel
+    store_parameters::Union{Nothing, ModelStoreParams}
+    ext::Dict{String, Any}
+end
+
+function ModelInternal(
+    container::T;
+    ext = Dict{String, Any}(),
+    recorders = [],
+) where {T <: AbstractOptimizationContainer}
+    return ModelInternal{T}(
+        container,
+        nothing,
+        BuildStatus.EMPTY,
+        RunStatus.READY,
+        true,
+        1, #Default executions is 1. The model will be run at least once
+        0,
+        nothing,
+        nothing,
+        Dict{TimeSeriesCacheKey, TimeSeriesCache}(),
+        recorders,
+        Logging.Warn,
+        Logging.Info,
+        nothing,
+        ext,
+    )
+end
+
+function add_recorder!(internal::ModelInternal, recorder::Symbol)
+    push!(internal.recorders, recorder)
+    return
+end
+
+get_recorders(internal::ModelInternal) = internal.recorders
+
+function configure_logging(internal::ModelInternal, file_mode)
+    return configure_logging(;
+        console = true,
+        console_stream = stderr,
+        console_level = internal.console_level,
+        file = true,
+        filename = joinpath(internal.output_dir, PROBLEM_LOG_FILENAME),
```

Comment:
`PROBLEM_LOG_FILENAME` isn’t defined.

Reply:
These should be defined by each package. I'll move it to be an argument